Merge 22.2-rc-3 to trunk

oguzkocer opened the pull request:
Contains the changes from https://github.com/wordpress-mobile/WordPress-Android/pull/18296, https://github.com/wordpress-mobile/WordPress-Android/pull/18301 & https://github.com/wordpress-mobile/WordPress-Android/pull/18290.

There were 3 conflicts in 2 files:

```
both modified:   WordPress/src/main/res/values-de/strings.xml
both modified:   build.gradle
```

---

```
<<<<<<< HEAD
    <string name="site_creation_domain_cost">%s/Jahr</string>
=======
    <string name="me_btn_help">Hilfe</string>
>>>>>>> origin/trunk
```

I kept both strings as they both seem to be in use.

---

```
<<<<<<< HEAD
    gutenbergMobileVersion = 'v1.93.1'
=======
    gutenbergMobileVersion = 'v1.94.0-alpha1'
>>>>>>> origin/trunk
```

Per usual, I kept the alpha version over the hotfix one.

---


```
<<<<<<< HEAD
    wordPressUtilsVersion = '3.6.1'
=======
    wordPressUtilsVersion = '3.6.0'
    indexosMediaForMobileVersion = '43a9026f0973a2f0a74fa813132f6a16f7499c3a'
>>>>>>> origin/trunk
```

For `wordPressUtilsVersion`, I kept the later version and `indexosMediaForMobileVersion` is just an addition.

irfano committed:
Use PermissionUtils functions everywhere for camera permission

irfano committed:
Update WordPress-Utils library to the PR

irfano committed:
Update WordPress-Utils version to the latest

Commit:
Update Gutenberg ref

ravishanker committed:
Fix dashboard card domain navigation

Pass domain_purchase parameter to list prices and not as free

Commit:
Update Gutenberg ref

Commit:
Update Gutenberg ref

Commit:
Merge pull request #18296 from wordpress-mobile/gutenberg/integrate_release_1.93.1

Integrate gutenberg-mobile release 1.93.1

irfano committed:
Merge pull request #18301 from wordpress-mobile/Dasboard-Card-Domains-Fix-Navigation

Fix dashboard card domain navigation

irfano committed:
Merge pull request #18290 from wordpress-mobile/fix/camera-permission-issue

Fix camera permission issue

oguzkocer committed:
Update translations

oguzkocer committed:
Bump version number

oguzkocer committed:
Merge remote-tracking branch 'origin/trunk' into merge/22.2-rc-3-to-trunk

ParaskP7 reviewed (approved):
👋 @oguzkocer !

FYI: There is a build error due to this `me_btn_help` string found more than once within `values-de` ([CI](https://buildkite.com/automattic/wordpress-android/builds/10781#0187b644-faf7-46de-8092-95292f52b9f9)). 🤷

<details>
    <summary>Build Failure</summary>

```
  | [2023-04-25T02:41:39Z] > Task :WordPress:mergeJetpackVanillaReleaseResources FAILED
  | [2023-04-25T02:41:39Z] ERROR:/var/lib/buildkite-agent/builds/ci-android-i-0c594a6ea7f9e0619-1/automattic/wordpress-android/WordPress/src/main/res/values-de/strings.xml: Resource and asset merger: Found item String/me_btn_help more than one time
  | [2023-04-25T02:41:39Z]
  | [2023-04-25T02:41:39Z] > Task :libs:editor:lintAnalyzeRelease
  | [2023-04-25T02:41:39Z] org.wordpress.android.lint.WordPressIssueRegistry in /var/lib/buildkite-agent/.gradle/caches/modules-2/files-2.1/org.wordpress/lint/1.1.0/d8905a4753b2053f888451b34448eb9ecc85e7de/lint-1.1.0.jar does not specify a vendor; see IssueRegistry#vendor
  | [2023-04-25T02:41:39Z]
  | [2023-04-25T02:42:01Z] FAILURE: Build failed with an exception.
  | [2023-04-25T02:42:01Z]
  | [2023-04-25T02:42:01Z] * What went wrong:
  | [2023-04-25T02:42:01Z] Execution failed for task ':WordPress:mergeJetpackVanillaReleaseResources'.
  | [2023-04-25T02:42:01Z] > /var/lib/buildkite-agent/builds/ci-android-i-0c594a6ea7f9e0619-1/automattic/wordpress-android/WordPress/src/main/res/values-de/strings.xml: Error: Found item String/me_btn_help more than one time
```

</details>

Otherwise, everything LGTM, included the merge conflict resolutions. As such, I'll approve this PR in order to unblock it from merging. Feel free to merge it as soon as the build failure gets resolved. 👍

oguzkocer committed:
Remove duplicate me_btn_help string from de strings

oguzkocer commented:
Thanks @ParaskP7 🙇‍♂️

oguzkocer committed:
Remove site_creation_domain_cost translation for de & es

The number of arguments for this string has been changed in #18315, so
these translations are no longer accurate and causing lint failures.

oguzkocer commented:
The number of arguments for `site_creation_domain_cost` has been updated which has resulted in a lint failure after cab6f766484e717273355631156c87210f42813e. I've tried to address this by removing the incorrect `site_creation_domain_cost` translations for `es` & `de` in 06d6645ffcc272b217265aee679ebd5f2102642c. 

I've also left a [suggestion in the original PR](https://github.com/wordpress-mobile/WordPress-Android/pull/18315#discussion_r1176944147) to update the string name along its value - especially when the number of arguments have changed.
